auth/src/main/java/com/firebase/ui/auth/util/OnMergeFailedListener.java

```diff
@@ -0,0 +1,11 @@
+package com.firebase.ui.auth.util;
+
+public interface OnMergeFailedListener {
```

Comment:
Building on my previous comment, I don't think we need to use the listener pattern for merge failures.  The only information returned on a merge failure is the previous user ID, so why not just do it through Activity results? The developer could handle it like this?

```java
@Override
public void onActivityResult(int requestCode, int resultCode, Intent data) {
  super.onActivityResult(requestCode, resultCode, data);

  if (requestCode == RC_SIGN_IN) {
    if (resultCode == RESULT_MERGE_FAILED) {
      String prevUid = data.getStringExtra(EXTRA_PREV_UID);
      // Handle merge failure ....
    }
  }
}
```

This also solves the issue you raised of needing to somehow store the listener reference in the FlowParams, which won't work (as you mentioned).